libcontainer/nsenter/nsenter_go121.go

```diff
@@ -0,0 +1,78 @@
+//go:build linux && !gccgo && go1.21
+// +build linux,!gccgo,go1.21
+
+package nsenter
+
+// Since Go 1.21 <https://github.com/golang/go/commit/c426c87012b5e>, the Go
+// runtime will try to call pthread_getattr_np(pthread_self()). This causes
+// issues with nsexec and requires backward compatibility with old versions
+// of the glibc.
+
+/*
+#cgo LDFLAGS: -Wl,--wrap=x_cgo_getstackbound
+#ifndef _GNU_SOURCE
+#  define _GNU_SOURCE
+#endif
+#include <stdint.h>
+#include <stdbool.h>
+#include <pthread.h>
+
+typedef uintptr_t uintptr;
+
+void
+__wrap_x_cgo_getstackbound(uintptr bounds[2])
+{
+	pthread_attr_t attr;
+	void *addr;
+	size_t size;
+#if defined(__GLIBC__) || defined(__illumos__)
+	bool noinit = false;
+	int err;
+#endif
+
+#if defined(__GLIBC__) || (defined(__sun) && !defined(__illumos__))
+	// pthread_getattr_np is a GNU extension supported in glibc.
+	// Solaris is not glibc but does support pthread_getattr_np
+	// (and the fallback doesn't work...). Illumos does not.
+	err = pthread_getattr_np(pthread_self(), &attr);  // GNU extension
+	if (err != 0) {
+		noinit = true;
+		goto failback;
+	}
+	err = pthread_attr_getstack(&attr, &addr, &size); // low address
+	if (err != 0) {
+		goto failback;
+	}
+	goto ret;
+#elif defined(__illumos__)
+	pthread_attr_init(&attr);
+	err = pthread_attr_get_np(pthread_self(), &attr);
+	if (err != 0) {
+		goto failback;
+	}
+	err = pthread_attr_getstack(&attr, &addr, &size); // low address
+	if (err != 0) {
+		goto failback;
+	}
+	goto ret;
+#endif
+	// We don't know how to get the current stacks, so assume they are the
+	// same as the default stack bounds.
+#if defined(__GLIBC__) || defined(__illumos__)
+failback:
+	if (!noinit) {
+		pthread_attr_destroy(&attr);
+	}
+#endif
+	pthread_attr_init(&attr);
```

Comment:
Guess you can reuse the `attr` initialized earlier, instead of doing destroy/init.

IOW
```C
fallback:
	if (noinit) {
		pthread_attr_init(&attr);
	}
```

Reply:
(and remove the whole #ifdef/endif, lines 61-66)

Reply:
Refactor now, it's outdated.